overlord/servicestate/quota_handlers.go

```diff
@@ -823,7 +854,75 @@ func validateSnapForAddingToGroup(st *state.State, snaps []string, group string,
 			}
 		}
 	}
+	return nil
+}
 
+// splitSnapServiceName splits and verifies the snap service reference
+// taken in by the frontend. It expects the format snap.service
+func splitSnapServiceName(name string) (string, string, error) {
+	parts := strings.Split(name, ".")
+	if len(parts) != 2 {
+		return "", "", fmt.Errorf("invalid snap service: %s", name)
+	}
+	if err := naming.ValidateSnap(parts[0]); err != nil {
+		return "", "", err
+	}
+	return parts[0], parts[1], naming.ValidateApp(parts[1])
+}
+
+// appReferenceIsService returns whether the service referred to in the
+// snap is actually a service.
+func appReferenceIsService(st *state.State, snap, service string) error {
+	snapInfo, err := snapstate.CurrentInfo(st, snap)
+	if err != nil {
+		return err
+	}
+
+	appInfo, ok := snapInfo.Apps[service]
+	if !ok {
+		return fmt.Errorf("invalid service %q", service)
+	}
+	if !appInfo.IsService() {
+		return fmt.Errorf("%q is not a service", service)
+	}
+	return nil
+}
+
+// ensureGroupHasNoSubgroups returns true if the group has no sub-groups or it doesn't exist,
+// otherwise turns false.
+func ensureGroupHasNoSubgroups(group string, allGrps map[string]*quota.Group) bool {
```

Comment:
Rename it to `groupHasSubGroups` (and swap the return value), since this is just returning an information.

Reply:
Ah makes a lot of sense, thank you